UI/ChatGPT/ChatGPT/ChatGPT/Services/ChatService.cs

```diff
@@ -0,0 +1,93 @@
+using ChatGPT.Business;
+using OpenAI.Managers;
+using OpenAI;
+using System.Runtime.CompilerServices;
+using OpenAI.ObjectModels.RequestModels;
+using OpenAI.ObjectModels;
+using System.Text;
+using OpenAI.ObjectModels.ResponseModels;
+
+namespace ChatGPT.Services;
+public class ChatService : IChatService
+{
+	private OpenAIService _client;
+
+	public ChatService(IOptions<AppConfig> appConfig)
+	{
+		var apiKey = appConfig.Value.ApiKey;
+
+		_client = new OpenAIService(
+			new OpenAiOptions()
+			{
+				ApiKey = apiKey
+			});
+	}
+
+	public async ValueTask<ChatResponse> AskAsync(string prompt)
+	{
+
+		var result = await _client.ChatCompletion.CreateCompletion(new ChatCompletionCreateRequest()
+		{
+			Messages = new List<ChatMessage>
+			{
+				ChatMessage.FromUser(prompt)
+			},
+			Model = Models.Gpt_3_5_Turbo
+		});
+
+		if (result.Successful)
+		{
+			var _result = result.Choices.Select(choice => choice.Message.Content);
+			return new ChatResponse(Status.Value, string.Join("", _result));
+		}
+		else
+		{
+			if (result.Error == null)
+			{
+				throw new Exception("Unknown Error");
+			}
```

Comment:
This will drive the message to stay in the UI with `...`. As we do have error info in the `ChatResponse` we should use it and not throw (we should even try/catch the whole API call to match the behavior of the streamed response).

Reply:
Should I add "Unknown Error" to the message in this case?

https://github.com/unoplatform/Uno.Samples/blob/58be3d9b161ce92418b61642cc32f0ec7b272dfa/UI/ChatGPT/ChatGPT/ChatGPT/Services/ChatService.cs#L50-L54

Reply:
Resolved.